src/file/serializer.cpp

```diff
@@ -0,0 +1,54 @@
+#include <QString>
+#include <QFile>
+
+#include "drawable/fractalCurve.hpp"
+#include "drawable/fractalLine.hpp"
+
+#include "drawable/drawable.hpp"
+#include "file/serializer.hpp"
+
+const quint32 MAGIC = 0x4c504d41; // hex LPMA
+
+namespace Lipuma{
+	void SerializeCanvas(Canvas* canvas, QString* filename){
+		QFile file(*filename);
+		if (!file.open(QIODevice::WriteOnly)){
+			qWarning("File failed!");
+		}else{
+			QDataStream stream(&file);
+			stream << MAGIC;
+			for (auto i : canvas->scene()->items()){
+				if (i->type() == Drawable::Type){
+					dynamic_cast<Drawable*>(i)->write(stream);
+				}
+			}
+			qWarning("File written.");
+		}
+	}
+
+	Canvas* LoadCanvas(QString* filename){
+		QFile file(*filename);
```

Comment:
A string reference rather than a pointer would inherently prevent null pointer issues here (references must generally be non-null).